inference/core/workflows/core_steps/classical_cv/image_preprocessing/v1.py

```diff
@@ -0,0 +1,220 @@
+                    "required": True,
+                },
+            },
+        },
+    )
+    rotation_degrees: Union[int, WorkflowParameterSelector(kind=[INTEGER_KIND])] = Field(  # type: ignore
+        title="Degrees of Rotation",
+        description="Positive value to rotate clockwise, negative value to rotate counterclockwise",
+        default=90,
+        examples=[90, "$inputs.rotation_degrees"],
+        gte=-360,
+        le=360,
+        json_schema_extra={
+            "relevant_for": {
+                "task_type": {
+                    "values": ["rotate"],
+                    "required": True,
+                },
+            }
+        },
+    )
+    flip_type: Union[WorkflowParameterSelector(kind=[STRING_KIND]), Literal["vertical", "horizontal", "both"]] = Field(  # type: ignore
+        title="Flip Type",
+        description="Type of flip to be applied to the image.",
+        default="vertical",
+        examples=["vertical", "horizontal", "$inputs.flip_type"],
+        json_schema_extra={
+            "relevant_for": {
+                "task_type": {
+                    "values": ["flip"],
+                    "required": True,
+                },
+            }
+        },
+    )
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(name="image", kind=[IMAGE_KIND]),
+        ]
+
+    @classmethod
+    def get_execution_engine_compatibility(cls) -> Optional[str]:
+        return ">=1.0.0,<2.0.0"
+
+
+class ImagePreprocessingBlockV1(WorkflowBlock):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @classmethod
+    def get_manifest(cls) -> Type[ImagePreprocessingManifest]:
+        return ImagePreprocessingManifest
+
+    def run(
+        self,
+        image: WorkflowImageData,
+        task_type: str,
+        width: Optional[int],
+        height: Optional[int],
+        rotation_degrees: Optional[int],
+        flip_type: Optional[str],
+        *args,
+        **kwargs,
+    ) -> BlockResult:
+
+        response_image = None
+
+        if task_type == "resize":
+            response_image = apply_resize_image(image.numpy_image, width, height)
+        elif task_type == "rotate":
+            response_image = apply_rotate_image(image.numpy_image, rotation_degrees)
+        elif task_type == "flip":
+            response_image = apply_flip_image(image.numpy_image, flip_type)
+        else:
+            raise ValueError(f"Invalid task type: {task_type}")
+
+        output_image = WorkflowImageData(
+            parent_metadata=image.parent_metadata,
+            workflow_root_ancestor_metadata=image.workflow_root_ancestor_metadata,
+            numpy_image=response_image,
+        )
+        return {"image": output_image}
+
+
+def apply_resize_image(
+    np_image: np.ndarray, width: Optional[int], height: Optional[int]
+) -> np.ndarray:
+    if width is None and height is None:
+        return np_image
```

Comment:
```suggestion
        return np_image.copy()
```

We should return copy of the image to be consistent with remainder of this method where copy is returned

Reply:
👍